mne/io/nicolet/nicolet.py

```diff
@@ -179,6 +179,7 @@ def __init__(self, input_fname, ch_type, montage=None, eog=(), ecg=(),
             info, preload, filenames=[input_fname], raw_extras=[header_info],
             last_samps=last_samps, orig_format='int',
             verbose=verbose)
+        self._init_kwargs = _get_argvalues()
```

Comment:
Since all our readers must call the `super` init, a cleaner way to do this is call `_get_argvalues` in `BaseRaw.__init__` and in `_get_argvalues` move up one stack frame before parsing

Reply:
I think you could make it work by moving up one stack before parsing but because `BaseRaw` is public, it could lead to unexpected/garbage in `raw._init_kwargs` if it was called by the user directly.

Reply:
I don't think we really need to worry about that corner case (at least on balance with the simplifications from the code we get by keeping it in `BaseRaw`).

Reply:
... but if you are sufficiently worried about it, you could add a check that the function/method in the desired stacklevel is in the `mne` namespace. If it's not, set `_init_kwargs = None` or so.